src/components/editor-content/editor-content.tsx

```diff
@@ -41,17 +149,199 @@ const EditorContent: React.FC<Props> = () => {
         </Typography>
       </>
     );
   }
 
+  /**
+   * Renders selector of filter scope
+   */
+  const renderSelectScope = () => {
+    return (
+      <>
+        <FormControl variant="outlined" className={ classes.selectScope }>
+          <TextField
+            select
+            id="scope-select-outlined"
+            size="small"
+            value={ scope }
+            onChange={ handleDateFormatChange }
+          >
+            { renderSelectOptions }
+          </TextField>
+        </FormControl>
+      </>
+    );
+  }
+
+  /**
+   * Renders start date picker 
+   * 
+   * @returns start date picker
+   */
+  const renderStartDatePicker = () => {
+    const { filterStartingDate } = strings.editorContent;
+
+    return (
+      <MuiPickersUtilsProvider utils={ DateFnsUtils } >
+        <Grid className={ classes.timeFilter }>
+          <KeyboardDatePicker
+            variant="inline"
+            views={ [ scope ] }
+            format={ dateFormat }
+            id="date-picker-start"
+            label={ filterStartingDate }
+            value={ selectedStartingDate }
+            onChange={ handleStartDateChange }
+            KeyboardButtonProps={ {"aria-label": `${ filterStartingDate }`} }
+          />
+        </Grid>
+      </MuiPickersUtilsProvider>
+    );
+  }
+  
+  /**
+   * Renders start year picker and week selector 
+   * 
+   * @returns start year picker and week selector
+   */
+  const renderStartYearPickerAndWeekSelector = () => {
+    return (
+      <>
+        <FormControl variant="standard">
+          <MuiPickersUtilsProvider utils={ DateFnsUtils } >
+            <Grid className={ classes.timeFilterYearSelector }>
+              <KeyboardDatePicker
+                variant="inline"
+                views={[ "year" ]}
+                format="yyyy"
+                id="date-picker-year-start"
+                label={ strings.editorContent.selectYearStart }
+                value={ selectedStartingDate }
+                onChange={ handleStartDateChange }
+                KeyboardButtonProps={ {"aria-label": `${ strings.editorContent.filterStartingDate }`} }
+              />
+            </Grid>
+          </MuiPickersUtilsProvider>
+        </FormControl>
+        <FormControl variant="standard" className={ classes.selectWeekNumbers }>
+          <FormHelperText>{ strings.editorContent.selectWeekStart }</FormHelperText>
+            <Select
+              labelId="scope-select-outlined-label"
+              id="scope-select-outlined"
+              value={ startWeek }
+              onChange={ handleStartWeekChange }
+            >
+              { generateWeekNumbers().map((weekNumber : number, index: number) => {
+                return  (
+                  <MenuItem 
+                    key={ index } 
+                    value={ weekNumber }
+                    >
+                    { weekNumber }
+                  </MenuItem>
+                )
+              }) }
```

Comment:
Move this to own rgeneric ender function, for example renderWeekNumbers and write the logic inside that. This function can then be used also in function renderEndYearPickerAndWeekSelector